packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1640,6 +1640,12 @@ export class TransactionController extends BaseControllerV1<
       updatedTransactionMeta,
       `TransactionController:updateCustodialTransaction - Custodial transaction updated`,
     );
+    if (
+      status &&
```

Comment:
Very minor, but I don't think we need the `status` check since the array won't contain `undefined` so the main condition should be enough.

Reply:
I thought the same but the main condition is just checking for the includes, the second one wouldn't be necessary if we throw if the status is `undefined` in the main condition. I can revisit the logic and refactor it to work around that.

Reply:
You can ignore the previous comment, done here https://github.com/MetaMask/core/pull/3845/commits/504f393bcc477ddb765efeaa9453db82228b3de3 .